### DISCUSSION TITLE
Passing Images From Memory

### DISCUSSION THREAD

**Discussion by erikhagendorn** [facebookresearch/detectron2#3052] [Q&A]
It appears images can only be passed as a file system path for the `file_name` key/value pair in the dataset dictionary.

How would I go passing image data directly from memory, without any files?

I am looking into using Detectron2 on large gigapixel images, in my typical pytorch workflow I have a custom `Dataset` which tiles and feeds the data from the larger image on the fly, I'm looking to do the same here.

**Comment by ExtReMLapin** [comment]
Did you find a solution for it ?

**Reply by 3R1K** [reply]
Unfortunately I did not.

**Comment by quantumdot** [comment]
Implement your own `DatasetMapper`. See https://detectron2.readthedocs.io/en/latest/tutorials/data_loading.html
